src/screens/MyWorkouts/WorkoutSearchBar.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { ScrollView, TextInput } from 'react-native-gesture-handler';
+import { supabase } from '../../utils/supabaseClient';
+
+/**
+ * @param selectedExercise.exercise //value set to exercise pressed
+ * @param selectedExercise.isOk //if true, then value of exercise is valid, otherwise disregard
+ */
+export type SearchBarWorkoutProps = {
```

Comment:
We should have a callback prop like `onSelectExercise: (s: string) => void` that the parent component passes in, and then when an exercise is clicked, we call the callback, passing in the exercise name as the parameter.

Reply:
We actually don't need to pass in the `selectedExercise` into the search bar component. We want the search bar to be in charge of its own state. But to that point, the search bar doesn't really need to keep track of the selected exercise, because as soon as we click on an exercise, we want to call the callback `onSelectExercise` and pass the name back to the parent component and then we close this search bar.